Formula/notcurses.rb

```diff
@@ -0,0 +1,32 @@
+class Notcurses < Formula
+  desc "Blingful character graphics/TUI library"
+  homepage "https://nick-black.com/dankwiki/index.php/Notcurses"
+  url "https://github.com/dankamongmen/notcurses/archive/refs/tags/v2.3.12.tar.gz"
+  sha256 "ce042908fac11f7df1f9eaa610e46e9c615f53ab036b7c27ae2396292512407b"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "doctest" => :build
+  depends_on "pandoc" => :build
+  depends_on "pkg-config" => :build
+  depends_on "ffmpeg"
+  depends_on "libunistring"
+  depends_on "ncurses"
+  depends_on "readline"
+  uses_from_macos "zlib"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build"
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+  end
+
+  test do
+    # without the TERM definition, Notcurses is like a small child, lost in a
+    # horrible mall. of course, if the tests are run in some non-xterm
+    # environment, this choice might prove unfortunate.
+    # you have no chance to survive. make your time.
+    ENV["TERM"] = "xterm"
+    system "#{bin}/notcurses-info"
```

Comment:
Can we check some output here to make sure it's doing what we want?

Reply:
not really; that's free-form output, subject to time at any change. if we want a clear signal as to whether most properties of the package work, we ought IMO run `notcurses-tester` as originally specified. if we want to ensure the library can link and initialize without catching fire, `notcurses-info` as we have it now seems a good choice. if we want to check some random 10% of the library, we could use `notcurses-tester` with specified `--tc=` and `--etc=` (testcases and exclude test cases, respectively). all three of these options give a clear return value via their exit status.

i mean yes i can throw a grep in there that checks that the first line starts with `notcurses`, but that's really *just* testing (a) linkage and (b) initialization. but i can do it.

Reply:
Linking and initializing is really all we care about. This is a validation for Homebrew, not really for the software itself.
This would be a good option then:

```suggestion
    assert_match "notcurses", shell_output("#{bin}/notcurses-info")
```